src/Microsoft.Extensions.Globalization.CultureInfoCache/CultureInfoCache.cs

```diff
@@ -22,11 +24,12 @@ public static partial class CultureInfoCache
         /// A read-only cached <see cref="CultureInfo"/> or <c>null</c> a match wasn't found in
         /// <see cref="KnownCultureNames"/>.
         /// </returns>
-        public static CultureInfo GetCultureInfo(string name)
+        public static CultureInfo GetCultureInfo(string name, IList<CultureInfo> SupportedCultures)
         {
             // Allow only known culture names as this API is called with input from users (HTTP requests) and
             // creating CultureInfo objects is expensive and we don't want it to throw either.
-            if (name == null || !KnownCultureNames.Contains(name))
+            if (name == null || SupportedCultures == null ||
+                SupportedCultures.Where( s => s.Name.ToLowerInvariant() == name.ToLowerInvariant()).FirstOrDefault() == null)
```

Comment:
1. Formatting
2. Can save a copy of `name.ToLowerInvariant()` (it doesn't change)
3. `s` --> `supportedCulture`
4. Why call FirstOrDefault and compare to null? Is that just `!foo.Any()`?